### DISCUSSION TITLE
Do imports used in getStaticProps get bundled?

### DISCUSSION THREAD

**Discussion by mrclay** [vercel/next.js#18051] [Help] [answered]
I want to make sure my getStaticProps code does not end up in JS bundles, and that includes functions called which have to be imported at the page level.

Does using Typescript have any effect?

Do I need to use require() and/or node’s fs.readfileSync to load modules and any API keys inside the gSP function body?

**Comment by ljosberinn** [comment, marked as answer]
Using typescript has no effect on this and [as the docs mention](https://nextjs.org/docs/basic-features/data-fetching#write-server-side-code-directly), node code/code included in getStaticProps/getServerSideProps/getStaticPaths does not get bundled client side. You can easily confirm this e.g. if you enable `browserSourceMaps: true` in next.config.js > experimental and check the resulting source maps, which show any server side code greyed out.

**Reply by mrclay** [reply]
Thanks! I’ll verify. Again, I understand the gSP() function itself won’t be bundled. Concern is `import` in the containing module.

**Comment by mrclay** [comment]
https://next-code-elimination.now.sh/s/yDIpMw_Iw looks like I just need to be careful to avoid combining client and server API exports in the same module(s).

**Reply by ljosberinn** [reply]
Yes, that is indeed the case, sorry for not intitially mentioning it. I've built something for sentry [here](https://github.com/ljosberinn/personal-react-boilerplate/blob/master/src/utils/sentry/shared.ts) which boots only the relevant variant based on environment by passing the correct init function. 

As a result, only `client` and `shared` show up in the source maps:

![image](https://user-images.githubusercontent.com/29307652/96595093-99177d00-12eb-11eb-839c-2473980877b2.png)
